File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
       int n = nums.size();
        unordered_map<int, int>mp;

       for(int i=0;i<n;i++){
        int num = target - nums[i];
        if(mp.count(num))return{mp[num], i};
          mp[nums[i]] = i;
       }
       return {};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int [] ans = new int[2];
        int len =  nums.length;
       HashMap<Integer,Integer> mp = new HashMap<>();
       for(int i =0;i<len;i++){
         int diff = target - nums[i];
         if(mp.containsKey(diff)){
            return new int[]{mp.get(diff), i};
         }
         mp.put(nums[i],i);
       }
       return new int[]{-1,-1};
    }
}